.github/workflows/observer-e2e.yml

```diff
@@ -0,0 +1,108 @@
+name: Observer E2E
```

Comment:
I like this localnet focused testing, but not sure it needs to be in its own workflow. Maybe this should just be a new suite in the normal `ci.yml`.

Reply:
It makes sense.
The only argument pointing to having a separate file is that it can be run in parallel, and eventually it can work on separate machines making it faster and not conflicting with other localnet runs. Please consider, otherwise I am happy to merge into `ci.yml`

Reply:
Unit tests already take almost 20 minutes. They are the critical path.

Any addition of one minute unit/integration tests can add significant project costs.
Details: 50 weeks * 5 days * 10 builds * 20 developers / 60 minutes > 800 hours of additional waiting time for the project per year.

Reply:
our CI is configured to run all stages in parallel, so it should be fine

Reply:
All right then. I submitted an additional change addressing your findings.